Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py

```diff
@@ -0,0 +1,153 @@
+        tokens: List[Token] = list()
+        normalized_text = QueryProcessor.remove_diacritics(source)
+        # tokens.add
+        # tokens.add
+        # return Token.merge_all_tokens(tokens, source, self.extractor_type_name)
+        #return None
+        pass
+
+    def remove_ambiguous_time_zone(self, extract_result: List[ExtractResult]) -> List[ExtractResult]:
+        return [item for item in extract_result if self.config.ambiguous_time_zone_list in item.text]
+
+    def match_location_times(self, text: str, tokens: List[Token]) -> List[Token]:
+        result: List[Token] = list()
+
+        if not self.config.location_time_suffix_regex:
+            return result
+
+        time_match = regex.match(self.config.location_time_suffix_regex, text)
+
+        # Before calling a Find() in location matcher, check if all the matched suffixes by
+        # LocationTimeSuffixRegex are already inside tokens extracted by TimeZone matcher.
+        # If so, don't call the Find() as they have been extracted by TimeZone matcher, otherwise, call it.
+        is_all_suffix_inside_tokens = True
+        for match in time_match:
+            is_inside = False
+            for token in tokens:
+                if token.start <= match.index and token.end >= match.index + match.lenght:
+                    is_inside = True
+                    break
+                if not is_inside:
+                    is_all_suffix_inside_tokens = False
+
+                if not is_all_suffix_inside_tokens:
+                    break
+
+        if len(time_match) != 0 and not is_all_suffix_inside_tokens:
+            last_match_index = time_match[len(time_match)-1]
+            matches = self.config.location_matcher.find(text[0: last_match_index])
+            location_matches = MatchingUtil.remove_sub_matches(matches.find(text))
+
+            i = 0
+            for match in time_match:
+                has_city_before = False
+                while i < len(location_matches) and location_matches[i].end <= match.index:
+                    has_city_before = True
+                    i += 1
+
+                    if i == len(location_matches):
+                        break
+
+                if has_city_before and location_matches[i - 1].end == match.index:
+                    result.append(Token(location_matches[i - 1].start, match.index + match.lenght))
+                if i == len(location_matches):
+                    break
+
+        return result
+
+    def match_timezones(self, text: str) -> List[Token]:
+        result: List[Token] = list()
+        match: Match
+
+        # Direct UTC matches
+        if self.config.direct_utc_regex:
+            direct_utc = regex.finditer(self.config.direct_utc_regex, text)
+            for match in direct_utc:
+                result.append(Token(match.start(), match.end() + match.start()))
+
+            matches = self.config.time_zone_matcher.find(text)
```

Comment:
why are you using find() here? While find() retrieves an integer, finditer() returns an iterator of MatchObject instances. Can you review it?